docs/docs/api.md

```diff
@@ -387,8 +387,9 @@ When specifying the `command` field in your job hash, use `url-runner.bash` (mak
 
 ## Constraints
 
-These constraints will currently only work against attributes that are specifically set on the Mesos slaves [as described in the Mesos documentation](http://mesos.apache.org/documentation/latest/configuration).
-(i.e. the `hostname` attribute is not currently automatically available for constraints [as it is in Marathon](https://mesosphere.github.io/marathon/docs/constraints))
+These constraints will work against attributes that are specifically set on the Mesos slaves [as described in the Mesos documentation](http://mesos.apache.org/documentation/latest/configuration).
+
+If a `hostname` attribute is not explicitly specified, one will automatically be created and made available for constraints [as it is in Marathon](https://mesosphere.github.io/marathon/docs/constraints). It should be noted that calling out specific hostnames is not resilient to slave failure and should be avoided if possible.
```

Comment:
I am afraid that this is not have Marathon behaves.

If you look at [Marathon's Constraints.scala](https://github.com/mesosphere/marathon/blob/e3e867f00d4964d0367ec7e07519d6c9545681d6/src/main/scala/mesosphere/mesos/Constraints.scala#L34), you can see that it will always use the hostname taken from the `hostname` field in the offer, even if the offer also includes a `hostname` attribute.